Babylon/groups/azure/__init__.py

```diff
@@ -1,22 +1,32 @@
-from azure.identity import AzureCliCredential
-from azure.identity import DefaultAzureCredential
-from click import group
-from click import pass_context
+import sys
+import logging
+from azure.identity import AzureCliCredential, DefaultAzureCredential
+from azure.core.exceptions import ClientAuthenticationError
+from click import group, pass_context
 from click.core import Context
+from Babylon.utils.decorators import require_platform_key
 
 from .commands import list_commands
 from .groups import list_groups
 
+logger = logging.getLogger("Babylon")
 
 @group()
 @pass_context
-def azure(ctx: Context):
+@require_platform_key("api_scope", "api_scope")
+def azure(ctx: Context, api_scope: str):
     """Group allowing communication with Microsoft Azure Cloud"""
-    try:
-        credentials = AzureCliCredential()
-    except:
-        credentials = DefaultAzureCredential()
-    ctx.obj = credentials
+    for creds in [AzureCliCredential(), DefaultAzureCredential()]:
```

Comment:
Same comment as in `Babylon.groups.api` wecould only check credentials using `DefaultAzureCredential`